workspace-files/tools/file_operations.py

```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/env python3
+from dataclasses import dataclass
+from tools.load_text import load_text_from_workspace_file
+from tools.helper import setup_logger, get_openai_client
+
+import os
+import tiktoken
+import json
+from pathlib import Path
+from tools.gptscript_workspace import (
+    read_file_in_workspace,
+    write_file_in_workspace,
+    list_files_in_workspace,
+)
+
+logger = setup_logger(__name__)
+
+TIKTOKEN_MODEL = "gpt-4o"
+enc = tiktoken.encoding_for_model(TIKTOKEN_MODEL)
+TOKEN_THRESHOLD = 10000
+
+FILE_ENV = os.getenv("FILENAME", "")
+DIR_ENV = os.getenv("DIR", "")
+MAX_FILE_SIZE = 250_000
+
+UNSUPPORTED_FILE_TYPES = {
+    ".pdf",
+    ".docx",
+    ".doc",
+    ".pptx",
+    ".ppt",
+    ".xlsx",
+    ".xls",
+    ".jpg",
+    ".png",
+    ".gif",
+    ".mp3",
+    ".mp4",
+    ".zip",
+    ".rar",
+}
+
+
+@dataclass
+class Explain:
+    filename: str
+    selection: str
+
+
+@dataclass
+class Data:
+    prompt: str = ""
+    explain: Explain = None
+    improve: Explain = None
+    changedFiles: dict = None
+
+
+async def read_file(input_file: str):
+    """the enhanced workspace_read tool
+    This tool reads a file from the GPTScript workspace and returns the file content.
+    If the file has too many tokens, it summarizes the file content and returns the summary instead.
+
+    Raises:
+        ValueError: If the INPUT_FILE environment variable is not set
+        Exception: If the file content is not a valid UTF-8 encoded string
+    """
+
+    from tools.summarizer import (
+        DocumentSummarizer,
+        MODEL,
+        MAX_CHUNK_TOKENS,
+        MAX_WORKERS,
+    )
+
+    logger.info(f"Input file: {input_file}")
+    if not input_file:
+        raise ValueError("Error: INPUT_FILE environment variable is not set")
+
+    file_content: str = await load_text_from_workspace_file(input_file)
+    tokens = enc.encode(file_content)
+
+    # if the file has too many tokens, summarize it and return the summary
+    if len(tokens) > TOKEN_THRESHOLD:
```

Comment:
summarize when `len(tokens) > 10k` and otherwise return. we might want to change the threshold..?